src/python/bot/tasks/corpus_pruning_task.py

```diff
@@ -113,19 +112,22 @@ def _get_corpus_file_paths(corpus_path):
   ]
 
 
-def _limit_corpus_size(corpus_directory, size_limit):
+def _limit_corpus_size(corpus_url, size_limit):
   """Limit number of files in a corpus directory."""
-  files_list = os.listdir(corpus_directory)
+  files_list = list(storage.list_blobs_full(corpus_url))
   corpus_size = len(files_list)
 
   if corpus_size <= size_limit:
     # Corpus directory size is within limit, no more work to do.
     return
 
+  logs.log(
+      'Limit corpus at {corpus_url} from {corpus_size} to {size_limit}.'.format(
+          corpus_url=corpus_url, corpus_size=corpus_size,
+          size_limit=size_limit))
   files_to_delete = random.sample(files_list, corpus_size - size_limit)
   for file_to_delete in files_to_delete:
-    file_to_delete_full_path = os.path.join(corpus_directory, file_to_delete)
-    shell.remove_file(file_to_delete_full_path)
+    storage.delete(file_to_delete)
```

Comment:
I wonder how slow this is for large corpora. Another thought: is there some easy way of downloading just the selection of files we choose in random.sample? We'll delete the extra ones when we sync back anyway.

Reply:
Will be measured in gce env after this lands. i have some ideas on optimization, but can invest if we really need it. Right now, it seems to only happen on one fuzzer on internal. To download a selection of files, i can't of a clean way to hack into rsync_to_disk.